compiler/rustc_typeck/src/astconv/generics.rs

```diff
@@ -46,19 +48,44 @@ impl<'o, 'tcx> dyn AstConv<'tcx> + 'o {
         // Specific suggestion set for diagnostics
         match (arg, &param.kind) {
             (
-                GenericArg::Type(hir::Ty { kind: hir::TyKind::Path { .. }, .. }),
-                GenericParamDefKind::Const { .. },
+                GenericArg::Type(hir::Ty {
+                    kind: hir::TyKind::Path(rustc_hir::QPath::Resolved(_, path)),
```

Comment:
Ah, you know what, it might be that we lose some suggestions because of this increase in restriction, not because of the empty `Res::Err` match arm. Maybe you can pull this into the this match arm body and change the match in line 58 to match on `kind` instead of `path.res`.

Reply:
Actually this is what is getting hit for the lost suggestions, but I can add in an additional branch for unmatched things